experimental/transliteration/tests/cldr.rs

```diff
@@ -0,0 +1,71 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// TODO: find a way to keep cldr_testData uptodate
+
+use icu_provider::_internal::locid::Locale;
+use icu_transliteration::Transliterator;
+use std::collections::HashMap;
+use std::path::PathBuf;
+
+#[test]
+fn test_all_cldr() {
+    // broken BCP47 ids in CLDR
+    let map = HashMap::from([("und-t-d0-ascii", "und-t-und-latn-d0-ascii")]);
+
+    let mut in_order =
+        std::fs::read_dir(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/cldr_testData"))
+            .unwrap()
+            .map(|x| x.unwrap().path())
+            .collect::<Vec<_>>();
+    in_order.sort();
+    for path in in_order {
+        if path.ends_with("_readme.txt") {
+            continue;
+        }
+        let locale = path.file_stem().unwrap().to_str().unwrap();
+        let locale = map.get(locale).unwrap_or(&locale);
+        let locale: Locale = locale.parse().unwrap();
+        let t = Transliterator::try_new(locale.clone()).unwrap();
+        test_file(t, path);
+    }
+}
+
+#[track_caller]
```

Comment:
Doesn't make sense here. It reports the callsite _of this function_ instead of the line _in this function_ that causes a panic. Rust uses it for things like `.unwrap()`. In tests `#[track_caller]` is nice if there's a helper function that can panic and is called from multiple different `#[test]` functions.